packages/rpc-transport/src/apis/subscriptions/subscriptions-api.ts

```diff
@@ -0,0 +1,41 @@
+import { IRpcSubscriptionsApi, RpcSubscription } from '../../json-rpc-types';
+import { IRpcApiSubscriptions, RpcSubscriptionsApiConfig } from '../api-types';
+
+export function createJsonRpcSubscriptionsApi<TRpcSubscriptions extends IRpcApiSubscriptions>(
+    config?: RpcSubscriptionsApiConfig
+): IRpcSubscriptionsApi<TRpcSubscriptions> {
+    return new Proxy({} as IRpcSubscriptionsApi<TRpcSubscriptions>, {
+        defineProperty() {
+            return false;
+        },
+        deleteProperty() {
+            return false;
+        },
+        get<TNotificationName extends keyof IRpcSubscriptionsApi<TRpcSubscriptions>>(
+            ...args: Parameters<NonNullable<ProxyHandler<IRpcSubscriptionsApi<TRpcSubscriptions>>['get']>>
+        ) {
+            const [_, p] = args;
+            const notificationName = p.toString() as keyof TRpcSubscriptions as string;
+            return function (
+                ...rawParams: Parameters<
+                    TRpcSubscriptions[TNotificationName] extends CallableFunction
+                        ? TRpcSubscriptions[TNotificationName]
+                        : never
+                >
+            ): RpcSubscription<ReturnType<TRpcSubscriptions[TNotificationName]>> {
+                const params = config?.parametersTransformer
+                    ? config?.parametersTransformer(rawParams, notificationName)
+                    : rawParams;
+                const responseTransformer = config?.responseTransformer
+                    ? config?.responseTransformer<ReturnType<TRpcSubscriptions[TNotificationName]>>
+                    : (rawResponse: unknown) => rawResponse as ReturnType<TRpcSubscriptions[TNotificationName]>;
+                return {
+                    params,
+                    responseTransformer,
+                    subscribeMethodName: notificationName.replace(/Notifications$/, 'Subscribe'),
+                    unsubscribeMethodName: notificationName.replace(/Notifications$/, 'Unsubscribe'),
```

Comment:
This is the one little bit of Solana JSON RPC left in this general-purpose API creator. Another provider might not have their methods end in ‘Notifications’ and the RPC methods end in `Subscribe`/`Unsubscribe`. That's just a web3.js/Solana convention.

That is to say, for this to be _truly_ general purpose you have to let people supply `getSubscribeMethodName(notificationName): string` and `getUnsubscribeMethodName(notificationName): string`

Reply:
Yeah, this change should be added I think. Probably in this PR.

Reply:
Added this!